lib/src/generator/html_generator.dart

```diff
@@ -37,6 +37,52 @@ class HtmlGeneratorBackend extends GeneratorBackendBase {
   HtmlGeneratorBackend(
       super.options, super.templates, super.writer, super.resourceProvider);
 
+  @override
+  void generateClass(PackageGraph packageGraph, Library library, Class clazz) {
+    super.generateClass(packageGraph, library, clazz);
+    var data = ClassTemplateData(options, packageGraph, library, clazz);
+    var sidebarContent = templates.renderSidebarForContainer(data);
```

Comment:
Doesn't this undo the performance benefits of caching the sidebar?   We're going to be running the template for the sidebar repeatedly unless I'm misreading something.

Reply:
This shouldn't undo the caching benefits we had earlier (it should more-or-less mimic them): The problem we had _before_ caching is that each page would independently generate it's left sidebar (sibling elements) and right sidebar (children), so that the left sidebar was being generated n+1 times for n siblings (by each of the n siblings, and once by the parent). The caching mechanism ensured each sidebar was generated once. This change does something similar: each TemplateData object is responsible for generating its own right sidebar, containing is children elements.